dali/python/nvidia/dali/pipeline.py

```diff
@@ -1049,7 +1057,10 @@ def outputs(self):
             A list of `TensorList` objects for respective pipeline outputs
         """
         with self._check_api_type_scope(types.PipelineAPIType.SCHEDULED):
+            self._consumer_iter += 1
```

Comment:
Note, there's another way (😱) to consume outputs, named `share_outputs`. It's used by the iterators if you wrap the pipeline (and can be called manually too). 

On that note, I think we need tests with an iterator that wraps the pipeline.